src/sagas.js

```diff
@@ -163,37 +175,62 @@ function* fetchIcon ({ iconId }) {
 }
 
 function* refreshData (action) {
-  console.log('refreshData(): ', action.payload)
-  if (!action.payload.quiet) {
-    logDebug('refreshData(): not quiet')
+  logDebug('refreshData(): ', action.payload)
+
+  // refresh VMs and remove any that haven't been refreshed
+  const vmsFetchedById = []
+  for (let vmId of Selectors.getVmsFetchedById().toArray()) {
+    const status = yield fetchSingleVm(getSingleVm({ vmId }))
```

Comment:
Why fetching "by id" doesn't follow `action.payload.shallowFetch` however fetch "by count" does?

Reply:
Good question that I didn't spend a lot of time trying to unwind.

Really I think it can be reduced to saying that the fetch by page/count are intended to fetch the minimal info needed to support the card list and left navigation while the fetch by ID is intended to fetch all for a detail/edit page.  It's unfortunate that the store overlaps for these purposes.

It's something that deserves a bit of a deeper think.